addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/device/DSMRPort.java

```diff
@@ -0,0 +1,576 @@
+             * Due to the Dutch Smart Meter program where every residence is provided
+             * a smart for free and the meters are DSMR V4 or higher
+             * we assume the majority of meters communicate with HIGH_SPEED_SETTINGS
+             * For older meters this means initializing is taking probably 1 minute
+             */
+            portSettings = DSMRPortSettings.HIGH_SPEED_SETTINGS;
+        } else {
+            // Fixed port settings configured, use these
+            portSettings = fixedPortSettings;
+        }
+    }
+
+    /**
+     * Opens the Operation System Serial Port
+     * <p>
+     * This method opens the port and set Serial Port parameters according to
+     * the DSMR specification. Since the specification is clear about these
+     * parameters there are not configurable.
+     * <p>
+     * If there are problem while opening the port, it is the responsibility of
+     * the calling method to handle this situation (and for example close the
+     * port again).
+     * <p>
+     * Opening an already open port is harmless. The method will return
+     * immediately
+     *
+     * @return {@link DeviceStateDetail} containing the details about the DeviceState
+     */
+    public void open() {
+        synchronized (portLock) {
+            // Sanity check
+            if (portStatus.isOpen) {
+                logger.debug("Serial Port is already open, keep current port instance");
+
+                dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.OPENED);
+            }
+            try {
+                logger.debug("Opening port {}", portName);
+
+                // Opening Operating System Serial Port
+                CommPortIdentifier portIdentifier = CommPortIdentifier.getPortIdentifier(portName);
+                CommPort commPort = portIdentifier.open("org.openhab.binding.dsmr",
+                        (int) DSMRDeviceConstants.SERIAL_PORT_READ_TIMEOUT);
+                serialPort = (SerialPort) commPort;
+
+                // Configure Serial Port based on specified port speed
+                logger.debug("Configure serial port parameters: {}", portSettings);
+
+                if (portSettings != null) {
+                    serialPort.setSerialPortParams(portSettings.getBaudrate(), portSettings.getDataBits(),
+                            portSettings.getStopbits(), portSettings.getParity());
+
+                    // SerialPort is ready, open the reader
+                    logger.debug("SerialPort opened successful");
+                    try {
+                        serialPortInputStream = new BufferedInputStream(serialPort.getInputStream());
+                    } catch (IOException ioe) {
+                        logger.error("Failed to get inputstream for serialPort. Closing port", ioe);
+
+                        dsmrPortListener.handleDSMRPortEvent(DSMRPortEvent.ERROR);
+                    }
+                    logger.info("DSMR Port opened successful");
```

Comment:
might be worth to mention on which port you have successfully connected (there might be multiple).